includes/class-sensei-guest-user.php

```diff
@@ -164,6 +163,51 @@ public function sensei_set_current_user_to_none_if_not_open_course_related_actio
 		}
 	}
 
+	/**
+	 * Initializes the backend and admin actions for Guest users.
+	 *
+	 * @since $$next-version$$
+	 */
+	public static function init_guest_user_admin() {
+		add_filter( 'editable_roles', [ static::class, 'remove_guest_student_role' ], 11 );
+
+		add_action( 'pre_user_query', [ static::class, 'remove_guest_users' ], 11 );
+	}
+
+	/**
+	 * Remove Guest Student role from showing up Settings.
+	 *
+	 * @since $$next-version$$
+	 * @access private
+	 *
+	 *  @param array $roles List of roles.
+	 */
+	public static function remove_guest_student_role( $roles ) {
+		unset( $roles['guest_student'] );
+		return $roles;
+	}
+
+	/**
+	 * Remove guest users from user queries.
+	 *
+	 * @since $$next-version$$
+	 * @access private
+	 *
+	 *  @param WP_User_Query $query The user query.
+	 */
+	public static function remove_guest_users( WP_User_Query $query ) {
+		global $wpdb;
+
+		$query->query_where = str_replace(
```

Comment:
There is a `role__not_in` option that should take care of generating the meta query. But it needs to be hooked to `pre_user_query`, which runs before parsing the query vars.
```php
$query->query_vars['role__not_in'][] = self::ROLE;
```

The main thing I'm a bit worried about is performance, since we are adding this extra join to user queries not related to Sensei, affecting other parts of the site. I tested on a site with 100k users, and the main query in the Users screen goes from `0.0094s` to around `2.6603s`, about a 250x increase. For 1k, 10k sites it's less critical but still a significant overhead.

Looking for alternatives, the fastest option seems to be filtering for the `user_login` field:
`AND {$wpdb->users}.user_login NOT LIKE 'guest_user_%'` →  Time is `0.0131s`, which is just a 1.35x overhead. 

The tradeoff is a small loss in precision, since theoretically anyone could create a username matching that. Also tried using the `user_email` field but that's slower. To mitigate, we could add a `sensei_` prefix to the login name too, eg `sensei_guest_XXXX`.

Reply:
Thanks for the testing and benchmarks ❤️ Updated the code to use user_login here https://github.com/Automattic/sensei/pull/6318/commits/3a423b710225004e5b6fb1e27c3589ae4d2c96f8